corehq/blobs/mixin.py

```diff
@@ -50,7 +52,10 @@ def _blobdb_bucket(self):
         if self._id is None:
             raise ResourceNotFound(
                 "cannot manipulate attachment on unidentified document")
-        return join(_get_couchdb_name(type(self)), self._id)
+        identifier = self._id
+        if not SAFENAME.match(identifier):
+            identifier = u'unsafe-' + sha1(identifier.encode('utf-8')).hexdigest()
```

Comment:
It's possible that two docs could exist such that all of their blobs would end up in the same "bucket": if the `_id` of one doc is `unsafe-71fde39aef7bdffb97ed1fb4e39501cea385b036` and the `_id` of another is `uuid:123456`, for example. I don't think this is likely or a major issue, but happy for someone else to suggest a different/better design or point out why this is bad.